parse_xml.py

```diff
@@ -21,115 +22,207 @@ def changeExt(fname, ext):
     seq = (os.path.splitext(fname)[0], ext)
     return '.'.join(seq)
 
-#get field(date) and return in a proper iso format
-def formatDate(x, field):
-    date = dateutil.parser.parse(x.pop(field))
+#validate filetype argument
+def validType(s):
+    if s not in ("p","g","pt"):
+        msg = "Not a valid file type value: '{0}'.".format(s)
+        raise argparse.ArgumentTypeError(msg)
+    else:
+        return s
+
+#validate date
+def validDate(s):
+    for fmt in ("%Y%m%d","%Y"):
+        try:
+            datetime.strptime(s, fmt)
+            return s
+        except ValueError:
+            pass
+    msg = "Not a valid date: '{0}'.".format(s)
+    raise argparse.ArgumentTypeError(msg)
+
+#replace key in dictionary
+def replaceKey(data,newval,oldval):
```

Comment:
You don't need to fix this now, but stylistically, Python code should have commas followed by a space. The style with no spaces would set off an error in most style checkers (but [apparently it's not actually specified in PEP 8](http://stackoverflow.com/questions/31741418/pep8-space-after-a-comma)).